src/3d/CMakeLists.txt

```diff
@@ -239,6 +239,8 @@ set (QGIS_3D_RCCS  shaders.qrc  ../../resources/3d/textures/textures.qrc)
 
 add_library(qgis_3d SHARED ${QGIS_3D_SRCS} ${QGIS_3D_HDRS} ${QGIS_3D_RCCS} ${QGIS_3D_PRIVATE_HDRS})
 
+target_precompile_headers(qgis_3d PRIVATE $<$<COMPILE_LANGUAGE:CXX>:${CMAKE_SOURCE_DIR}/src/core/qgis.h>)
```

Comment:
Would this also work? Seems to be more DRY

```
target_precompile_headers(qgis_3d REUSE_FROM qgis_core)
```

Reply:
> `target_precompile_headers(qgis_3d REUSE_FROM qgis_core)`

no, that won't work in practice, because it would require that qgis_3d uses the *exact* same set of include directories, compiler options and warnings and compile definitions than qgis_core. If not, the compiler will not use the precompiled header and actually loose some time realizing it can't use it.
The cost of building the precompiled header for each target is about 4 seconds per target. Given that the build time is of each target is of the order of at least 30 minutes to several hours, that's very quickly amortized...